src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryContext.java

```diff
@@ -162,6 +165,14 @@ public String getName() {
     return rule.getName();
   }
 
+  @StarlarkMethod(
+      name = "build_workspace_directory",
```

Comment:
this concept is usually referred to as the "workspace root". Could you update the name throughout this PR?

Reply:
For what it's worth, `build_workspace_directory` makes sense to me and matches the related environment variable that's set on executable targets:
https://docs.bazel.build/versions/main/user-manual.html#run
> The following extra environment variables are also available to the binary:
>
>    `BUILD_WORKSPACE_DIRECTORY`: the root of the workspace where the build was run.
>   ` BUILD_WORKING_DIRECTORY`: the current working directory where Bazel was run from.

Reply:
Done - thanks!

Reply:
sorry to be nitpicky -- is there a reason why "build" is in the name? I kind of thought `ctx.workspace_root` would work well.

Also thanks @UebelAndre for the context -- I wasn't aware of the env variables. It makes more sense that those names include "BUILD", since env variables aren't Bazel-specific; but with the `ctx` object, we're obviously in Bazel.

Reply:
@meteorcloudy Oh boy, `Label.workspace_root` has a completely different meaning, though... I think we should deprecate the `Label.workspace_*` fields and replace them with `Label.repo_*` or something. The terms "repo" and "workspace" used to be used interchangeably but now we mostly think of the "workspace" as the overall space we work in, housing the main repo and the various external repos.

Reply:
Removed the build :)

Reply:
> Label.workspace_root has a completely different meaning, though...

Indeed, it's actually the workspace root of the repo of the label, `repo_root` is much better.